### PR TITLE
Add Slack incoming webhook detection and validator.

### PR DESCRIPTION
**Pull request by michaelcfanning**
@eddynaka @jameswinkler

### CODE REVIEW

File: Src/Plugins/Tests.Security/Tests.Security.csproj [hunk 1]
```diff
@@ -13,6 +13,11 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <ItemGroup>
```
Comment (jameswinkler) [inline Src/Plugins/Tests.Security/Tests.Security.csproj:16]: I don't think we need this group.

**Review by jameswinkler — APPROVED** [review]
:shipit:

**Review by v-luisangell — APPROVED** [review]

File: Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json [hunk 2]
```diff
@@ -83,10 +83,16 @@
...
+          "Id": "SEC101/012",
+          "Name": "DoNotExposePlaintextSecrets/SlackWebhook",
+          "ContentsRegex": "$SEC101/012.SlackWebhook",
+          "MessageArguments": { "secretKind": "Slack web hook" }
+        },
         {
           "Id": "SEC101/014",
           "Name": "DoNotExposePlaintextSecrets/FacebookAccessToken",
-          "ContentsRegex": "\\b(?<refine>EAACEdEose0cBA(?i)[0-9a-z]+)",
+          "ContentsRegex": "SEC101/104.FacebookAccessToken",
```
Comment (eddynaka) [inline Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json:95]: >SEC101/104 [](start = 28, length = 10)

fixed wrong id without $

File: Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json [hunk 3]
```diff
@@ -83,10 +83,16 @@
...
+        {
+          "Id": "SEC101/012",
+          "Name": "DoNotExposePlaintextSecrets/SlackWebhook",
+          "ContentsRegex": "$SEC101/012.SlackWebhook",
+          "MessageArguments": { "secretKind": "Slack web hook" }
+        },
         {
           "Id": "SEC101/014",
           "Name": "DoNotExposePlaintextSecrets/FacebookAccessToken",
-          "ContentsRegex": "\\b(?<refine>EAACEdEose0cBA(?i)[0-9a-z]+)",
```
Comment (eddynaka) [inline Src/Plugins/Security/SEC101.SecurePlaintextSecrets.json:89]: >\\b(?<refine>EAACEdEose0cBA(?i)[0-9a-z]+) [](start = 28, length = 41)

fixed missing regex in txt file

File: Src/Plugins/Tests.Security/Tests.Security.csproj (hunk 1 above)
Comment (eddynaka) [reply Src/Plugins/Tests.Security/Tests.Security.csproj:16]: fixed.

---
In reply to: [578079080](https://github.com/microsoft/sarif-pattern-matcher/pull/228#discussion_r578079080) [](ancestors = 578079080)

**Review by eddynaka — APPROVED** [review]
:shipit:

**Pull request merged**